src/main/scala/csp/Solver.scala

```diff
@@ -35,7 +35,24 @@ case class Solver(oscarModel: Model, bandit: String) {
       sn.swap()
     )
 
-    val restartN: Neighborhood = simpleNeighborhoods.shuffle(indices = Some(oscarModel.mostViolatedCars))
+    val mostViolated = oscarModel.mostViolatedCars
+    val violated     = oscarModel.violatedCars
+
+    val restart1: Neighborhood =
+      sn.shuffle(indices = Some(mostViolated)) guard (() =>
+        mostViolated.value.size > 2
+      )
+
+    val restart2: Neighborhood = sn.shuffle(
+      indices = Some(violated),
+      numOfPositions = Some(5 max violated.value.size / 2)
+    )
+
+    val restart3: Neighborhood = sn.shuffle(
+      numOfPositions = Some(oscarModel.instance.nCars / 2)
+    )
+
+    val restart4: Neighborhood = sn.shuffle()
```

Comment:
`restart4` is never used, should we remove it?

Reply:
As mentioned, I'd just keep it on the bench for now